react-frontend/src/profile/ProfilePage.js

```diff
@@ -88,6 +95,60 @@ function ProfilePage() {
     return phoneNumber;
   };
 
+
+  const handleDeleteConfirmation = async () => {
+    const token = localStorage.getItem('token');
+    if (!token) {
+      navigate('/login');
+      return;
+    }
+
+    try {
+      const response = await axios.delete(process.env.REACT_APP_BACKEND_LINK + `/users/delete`, {
+        headers: {
+          'Authorization': `Bearer ${token}`
+        },
+        data: {
+          username: deleteConfirmationData.username,
+          password: deleteConfirmationData.password
+        }
+      });
+
+      if (response.status === 200) {
+        // After successful deletion, redirect to login
+        localStorage.removeItem('token');
+        navigate('/login');
+      }
+    } catch (error) {
+      console.error('Failed to delete profile:', error.response);
+      if (error.response && error.response.data && error.response.data.error) {
+        setDeleteError(error.response.data.error);
+      } else {
+        setDeleteError('An error occurred while deleting the profile.');
+      }
+    }
+  };
+
+  const confirmDelete = () => {
```

Comment:
Function is used only once; could be redefined as lambda if that use case is still valid.

Reply:
Same thing with the rest of the deleting functions since they're only being used once.

Reply:
> Same thing with the rest of the deleting functions since they're only being used once.

Gotcha, I'll look into it